rclcpp/test/rclcpp/test_parameter_events_subscriber.cpp

```diff
@@ -0,0 +1,348 @@
+// Copyright 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+
+#include "gtest/gtest.h"
+#include "rclcpp/rclcpp.hpp"
+
+class TestParameterEventsSubscriber : public rclcpp::ParameterEventsSubscriber
+{
+public:
+  explicit TestParameterEventsSubscriber(rclcpp::Node::SharedPtr node)
+  : ParameterEventsSubscriber(node)
+  {}
+
+  void test_event(const rcl_interfaces::msg::ParameterEvent::SharedPtr event)
+  {
+    event_callback(event);
+  }
+};
+
+class TestNode : public ::testing::Test
+{
+protected:
+  static void SetUpTestCase()
+  {
+    rclcpp::init(0, nullptr);
+  }
+
+  void SetUp()
+  {
+    rclcpp::NodeOptions options;
+    options.allow_undeclared_parameters(true);
```

Comment:
Is this necessary? I looked through the code looking for `get/set_parameter` like functions being called on the node and I didn't see any.